### PR TITLE
Hw11 task1

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: test/hw11/test_task1.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+)
+def test_simplified_enum_metaclass_works_fine_for_class(cls, attr):
+    assert getattr(cls, attr) == attr
+
+
+@pytest.mark.parametrize(
+    ("cls", "attr"),
+    [
+        (ColorsEnum, "O"),
+        (SizesEnum, "O"),
```
Comment (Cjkjvfnby) [inline test/hw11/test_task1.py]: What is the difference between these tests?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: test/hw11/test_task1.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        "ORANGE",
+    ],
+)
+def test_simplified_enum_metaclass_works_fine_for_class(attr):
+    assert getattr(ColorsEnum, attr) == attr
+
+
+def test_simplified_enum_metaclass__if_attr_does_not_exist():
+    with pytest.raises(AttributeError, match="Does not exist:"):
+        _ = ColorsEnum.O
```
Comment (Cjkjvfnby) [inline test/hw11/test_task1.py]: `_ =` why do you need it?

File: test/hw11/test_task1.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    class Test(metaclass=SimplifiedEnum):
+        A = "A"
+
+        def __init__(self, a):
+            self.__keys = a
+
+    test = Test(0)
+    assert Test.A == "A"
+    assert test.A == "A"
+    assert test._Test__keys == 0
```
Comment (Cjkjvfnby) [inline test/hw11/test_task1.py]: What exactly you are testing here?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: test/hw11/test_task1.py (hunk 2 above)
Comment (E1-Bart0) [reply test/hw11/test_task1.py]: For me, it's more obvious to do this. And PyCharm doesn't blame me.

File: test/hw11/test_task1.py (hunk 3 above)
Comment (E1-Bart0) [reply test/hw11/test_task1.py]: That we couldn't overwrite `__keys`

Comment (Cjkjvfnby) [reply test/hw11/test_task1.py]: Where is the code that protects __keys?

File: test/hw11/test_task1.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import pytest
+
+from All_home_works.hw11.task1 import SimplifiedEnum
+
+
+class ColorsEnum(metaclass=SimplifiedEnum):
+    RED = "RED"
+    BLUE = "BLUE"
+    ORANGE = "ORANGE"
+    BLACK = "BLACK"
```
Comment (Cjkjvfnby) [inline test/hw11/test_task1.py]: What is the minimum number of items in the enum to test this feature?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: test/hw11/test_task1.py (hunk 4 above)
Comment (E1-Bart0) [reply test/hw11/test_task1.py]: To test it 1 item needed, but it's better to use at least 2 items.

Comment (Cjkjvfnby) [reply test/hw11/test_task1.py]: 2 looks reasonable.  Let' simplify the code.

File: test/hw11/test_task1.py (hunk 3 above)
Comment (E1-Bart0) [reply test/hw11/test_task1.py]: I didn't understand the question. But `__keys` are in metaclass.

Comment (Cjkjvfnby) [reply test/hw11/test_task1.py]: Why we couldn't overwrite __keys?

Comment (E1-Bart0) [reply test/hw11/test_task1.py]: Because it's a private attribute that references to metaclass.
`Test._SimplifiedEnum__keys`

Comment (Cjkjvfnby) [reply test/hw11/test_task1.py]: Is it the responsibility of your code to protect private variables?

Comment (E1-Bart0) [reply test/hw11/test_task1.py]: Of course, it's not

Comment (Cjkjvfnby) [reply test/hw11/test_task1.py]: Do we need to test this feature if you did not write it?

Comment (E1-Bart0) [reply test/hw11/test_task1.py]: Okey. I got you. It was in education purpose, will remove this test.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**